### ISSUE TITLE
CGAL 5 compile problems

### ISSUE THREAD

**Issue by varenius** [varenius/ascot#31]
Running "make" for "ivg" fails on Linux Mint 20 with libgcal-dev 5.0.2-3 (instead of 4.4. before). 

```
eskil@grotte:/opt/ascot/src/ivg$ make
g++   -c ./matrix.cpp -I/include -I./ -I/opt/ascot//include -I/opt/ascot//src/projectpluto/include -I/opt/ascot//src/iers -I/opt/ascot//src/sofa/include
./matrix.cpp: In member function ‘ivg::Matrix& ivg::Matrix::operator=(const std::vector<int>&)’:
./matrix.cpp:211:1: warning: no return statement in function returning non-void [-Wreturn-type]
  210 |     _cols = 1;
  +++ |+    return *this;
  211 | }
      | ^
g++  -c ./funother.cpp -I./
g++ -O3 -std=c++0x  -g -fPIC -c ./date.cpp   -I/include -I./ -I/opt/ascot//include -I/opt/ascot//src/projectpluto/include -I/opt/ascot//src/iers -I/opt/ascot//src/sofa/include
g++ -O3 -std=c++0x  -g -fPIC -o auxfunc.o -c ./auxfunc.cpp -I/include -I./ -I/opt/ascot//include -I/opt/ascot//src/projectpluto/include -I/opt/ascot//src/iers -I/opt/ascot//src/sofa/include
./auxfunc.cpp: In function ‘libconfig::Setting& get_list_element(libconfig::Setting&, std::string, int)’:
./auxfunc.cpp:117:1: warning: control reaches end of non-void function [-Wreturn-type]
  117 | }
      | ^
g++ -c -o tictoc.o tictoc.h
gcc -O3 -c ./sparse2array.c
g++ -O3 -std=c++0x  -g -fPIC -c ./eop_series.cpp   -I/include -I./ -I/opt/ascot//include -I/opt/ascot//src/projectpluto/include -I/opt/ascot//src/iers -I/opt/ascot//src/sofa/include
In file included from /usr/include/CGAL/Kernel/global_functions.h:25,
                 from /usr/include/CGAL/Cartesian/Cartesian_base.h:22,
                 from /usr/include/CGAL/Simple_cartesian.h:20,
                 from /usr/include/CGAL/Exact_predicates_inexact_constructions_kernel.h:20,
                 from ./trf.h:4,
                 from ./parser.h:10,
                 from ./eop_series.h:13,
                 from ./eop_series.cpp:1:
/usr/include/CGAL/Kernel/hash_functions.h:25:13: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
   25 | inline std::enable_if_t<std::is_same<typename K::Rep_tag, Cartesian_tag>::value, std::size_t>
      |             ^~~~~~~~~~~
      |             enable_if
/usr/include/CGAL/Kernel/hash_functions.h:47:13: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
   47 | inline std::enable_if_t<std::is_same<typename K::Rep_tag, Cartesian_tag>::value, std::size_t>
      |             ^~~~~~~~~~~
      |             enable_if
/usr/include/CGAL/Kernel/hash_functions.h:57:13: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
   57 | inline std::enable_if_t<std::is_same<typename K::Rep_tag, Cartesian_tag>::value, std::size_t>
      |             ^~~~~~~~~~~
      |             enable_if
/usr/include/CGAL/Kernel/hash_functions.h:66:13: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
   66 | inline std::enable_if_t<std::is_same<typename K::Rep_tag, Cartesian_tag>::value, std::size_t>
      |             ^~~~~~~~~~~
      |             enable_if
/usr/include/CGAL/Kernel/hash_functions.h:75:13: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
   75 | inline std::enable_if_t<std::is_same<typename K::Rep_tag, Cartesian_tag>::value, std::size_t>
      |             ^~~~~~~~~~~
      |             enable_if
/usr/include/CGAL/Kernel/hash_functions.h:84:13: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
   84 | inline std::enable_if_t<std::is_same<typename K::Rep_tag, Cartesian_tag>::value, std::size_t>
      |             ^~~~~~~~~~~
      |             enable_if
/usr/include/CGAL/Kernel/hash_functions.h:93:13: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
   93 | inline std::enable_if_t<std::is_same<typename K::Rep_tag, Cartesian_tag>::value, std::size_t>
      |             ^~~~~~~~~~~
      |             enable_if
/usr/include/CGAL/Kernel/hash_functions.h:102:13: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
  102 | inline std::enable_if_t<std::is_same<typename K::Rep_tag, Cartesian_tag>::value, std::size_t>
      |             ^~~~~~~~~~~
      |             enable_if
/usr/include/CGAL/Kernel/hash_functions.h:126:13: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
  126 | inline std::enable_if_t<std::is_same<typename K::Rep_tag, Cartesian_tag>::value, std::size_t>
      |             ^~~~~~~~~~~
      |             enable_if
/usr/include/CGAL/Kernel/hash_functions.h:135:13: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
  135 | inline std::enable_if_t<std::is_same<typename K::Rep_tag, Cartesian_tag>::value, std::size_t>
      |             ^~~~~~~~~~~
      |             enable_if
/usr/include/CGAL/Kernel/hash_functions.h:145:13: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
  145 | inline std::enable_if_t<std::is_same<typename K::Rep_tag, Cartesian_tag>::value, std::size_t>
      |             ^~~~~~~~~~~
      |             enable_if
/usr/include/CGAL/Kernel/hash_functions.h:154:13: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
  154 | inline std::enable_if_t<std::is_same<typename K::Rep_tag, Cartesian_tag>::value, std::size_t>
      |             ^~~~~~~~~~~
      |             enable_if
/usr/include/CGAL/Kernel/hash_functions.h:164:13: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
  164 | inline std::enable_if_t<std::is_same<typename K::Rep_tag, Cartesian_tag>::value, std::size_t>
      |             ^~~~~~~~~~~
      |             enable_if
/usr/include/CGAL/Kernel/hash_functions.h:174:13: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
  174 | inline std::enable_if_t<std::is_same<typename K::Rep_tag, Cartesian_tag>::value, std::size_t>
      |             ^~~~~~~~~~~
      |             enable_if
In file included from /usr/include/CGAL/Lazy_exact_nt.h:33,
                 from /usr/include/CGAL/internal/Exact_type_selector.h:25,
                 from /usr/include/CGAL/Exact_kernel_selector.h:26,
                 from /usr/include/CGAL/Filtered_kernel.h:22,
                 from /usr/include/CGAL/Exact_predicates_inexact_constructions_kernel.h:21,
                 from ./trf.h:4,
                 from ./parser.h:10,
                 from ./eop_series.h:13,
                 from ./eop_series.cpp:1:
/usr/include/CGAL/Lazy.h:108:30: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
  108 | template<class T>inline std::enable_if_t<std::is_arithmetic<T>::value||std::is_enum<T>::value, T> approx(T d){return d;}
      |                              ^~~~~~~~~~~
      |                              enable_if
/usr/include/CGAL/Lazy.h:109:30: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
  109 | template<class T>inline std::enable_if_t<std::is_arithmetic<T>::value||std::is_enum<T>::value, T> exact (T d){return d;}
      |                              ^~~~~~~~~~~
      |                              enable_if
/usr/include/CGAL/Lazy.h:110:30: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
  110 | template<class T>inline std::enable_if_t<std::is_arithmetic<T>::value||std::is_enum<T>::value, unsigned> depth(T){return 0;}
      |                              ^~~~~~~~~~~
      |                              enable_if
/usr/include/CGAL/Lazy.h:113:30: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
  113 | template<class T>inline std::enable_if_t<std::is_empty<T>::value, T> exact(T){return {};}
      |                              ^~~~~~~~~~~
      |                              enable_if
/usr/include/CGAL/Lazy.h:114:30: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
  114 | template<class T>inline std::enable_if_t<std::is_empty<T>::value, T> approx(T){return {};}
      |                              ^~~~~~~~~~~
      |                              enable_if
/usr/include/CGAL/Lazy.h:115:30: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
  115 | template<class T>inline std::enable_if_t<std::is_empty<T>::value, unsigned> depth(T){return 0;}
      |                              ^~~~~~~~~~~
      |                              enable_if
/usr/include/CGAL/Lazy.h:118:31: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
  118 | template <class T, class=std::enable_if_t<is_iterator_type<T,std::input_iterator_tag>::value>>
      |                               ^~~~~~~~~~~
      |                               enable_if
/usr/include/CGAL/Lazy.h:118:42: error: expected ‘>’ before ‘<’ token
  118 | template <class T, class=std::enable_if_t<is_iterator_type<T,std::input_iterator_tag>::value>>
      |                                          ^
/usr/include/CGAL/Lazy.h:119:1: error: ‘exact’ function uses ‘auto’ type specifier without trailing return type
  119 | auto exact(T const& t) {return make_transforming_iterator(t,[](auto const&u){return CGAL::exact(u);});}
      | ^~~~
/usr/include/CGAL/Lazy.h:119:1: note: deduced return type only available with ‘-std=c++14’ or ‘-std=gnu++14’
/usr/include/CGAL/Lazy.h: In function ‘auto CGAL::exact(const T&)’:
/usr/include/CGAL/Lazy.h:119:64: error: use of ‘auto’ in lambda parameter declaration only available with ‘-std=c++14’ or ‘-std=gnu++14’
  119 | auto exact(T const& t) {return make_transforming_iterator(t,[](auto const&u){return CGAL::exact(u);});}
      |                                                                ^~~~
/usr/include/CGAL/Lazy.h: In lambda function:
/usr/include/CGAL/Lazy.h:119:98: error: no matching function for call to ‘exact(const int&)’
  119 | auto exact(T const& t) {return make_transforming_iterator(t,[](auto const&u){return CGAL::exact(u);});}
      |                                                                                                  ^
/usr/include/CGAL/Lazy.h:84:1: note: candidate: ‘template<class AT, class ET, class E2A> const ET& CGAL::exact(const CGAL::Lazy<AT, ET, E2A>&)’
   84 | exact(const Lazy<AT,ET,E2A>& l)
      | ^~~~~
/usr/include/CGAL/Lazy.h:84:1: note:   template argument deduction/substitution failed:
/usr/include/CGAL/Lazy.h:119:98: note:   mismatched types ‘const CGAL::Lazy<AT, ET, E2A>’ and ‘const int’
  119 | auto exact(T const& t) {return make_transforming_iterator(t,[](auto const&u){return CGAL::exact(u);});}
      |                                                                                                  ^
/usr/include/CGAL/Lazy.h:104:1: note: candidate: ‘const CGAL::Bbox_2& CGAL::exact(const CGAL::Bbox_2&)’
  104 | CGAL_LAZY_FORWARD(Bbox_2)
      | ^~~~~~~~~~~~~~~~~
/usr/include/CGAL/Lazy.h:104:1: note:   no known conversion for argument 1 from ‘const int’ to ‘const CGAL::Bbox_2&’
  104 | CGAL_LAZY_FORWARD(Bbox_2)
      | ^~~~~~~~~~~~~~~~~
/usr/include/CGAL/Lazy.h:105:1: note: candidate: ‘const CGAL::Bbox_3& CGAL::exact(const CGAL::Bbox_3&)’
  105 | CGAL_LAZY_FORWARD(Bbox_3)
      | ^~~~~~~~~~~~~~~~~
/usr/include/CGAL/Lazy.h:105:1: note:   no known conversion for argument 1 from ‘const int’ to ‘const CGAL::Bbox_3&’
  105 | CGAL_LAZY_FORWARD(Bbox_3)
      | ^~~~~~~~~~~~~~~~~
/usr/include/CGAL/Lazy.h:119:6: note: candidate: ‘template<class T, class> auto CGAL::exact(const T&)’
  119 | auto exact(T const& t) {return make_transforming_iterator(t,[](auto const&u){return CGAL::exact(u);});}
      |      ^~~~~
/usr/include/CGAL/Lazy.h:119:6: note:   template argument deduction/substitution failed:
/usr/include/CGAL/Lazy.h: At global scope:
/usr/include/CGAL/Lazy.h:120:31: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
  120 | template <class T, class=std::enable_if_t<is_iterator_type<T,std::input_iterator_tag>::value>>
      |                               ^~~~~~~~~~~
      |                               enable_if
/usr/include/CGAL/Lazy.h:120:42: error: expected ‘>’ before ‘<’ token
  120 | template <class T, class=std::enable_if_t<is_iterator_type<T,std::input_iterator_tag>::value>>
      |                                          ^
/usr/include/CGAL/Lazy.h:121:1: error: ‘approx’ function uses ‘auto’ type specifier without trailing return type
  121 | auto approx(T const& t) {return make_transforming_iterator(t,[](auto const&u){return CGAL::approx(u);});}
      | ^~~~
/usr/include/CGAL/Lazy.h:121:1: note: deduced return type only available with ‘-std=c++14’ or ‘-std=gnu++14’
/usr/include/CGAL/Lazy.h: In function ‘auto CGAL::approx(const T&)’:
/usr/include/CGAL/Lazy.h:121:65: error: use of ‘auto’ in lambda parameter declaration only available with ‘-std=c++14’ or ‘-std=gnu++14’
  121 | auto approx(T const& t) {return make_transforming_iterator(t,[](auto const&u){return CGAL::approx(u);});}
      |                                                                 ^~~~
/usr/include/CGAL/Lazy.h: In lambda function:
/usr/include/CGAL/Lazy.h:121:100: error: no matching function for call to ‘approx(const int&)’
  121 | auto approx(T const& t) {return make_transforming_iterator(t,[](auto const&u){return CGAL::approx(u);});}
      |                                                                                                    ^
/usr/include/CGAL/Lazy.h:66:1: note: candidate: ‘template<class AT, class ET, class E2A> const AT& CGAL::approx(const CGAL::Lazy<AT, ET, E2A>&)’
   66 | approx(const Lazy<AT,ET,E2A>& l)
      | ^~~~~~
/usr/include/CGAL/Lazy.h:66:1: note:   template argument deduction/substitution failed:
/usr/include/CGAL/Lazy.h:121:100: note:   mismatched types ‘const CGAL::Lazy<AT, ET, E2A>’ and ‘const int’
  121 | auto approx(T const& t) {return make_transforming_iterator(t,[](auto const&u){return CGAL::approx(u);});}
      |                                                                                                    ^
/usr/include/CGAL/Lazy.h:75:1: note: candidate: ‘template<class AT, class ET, class E2A> AT& CGAL::approx(CGAL::Lazy<AT, ET, E2A>&)’
   75 | approx(Lazy<AT,ET,E2A>& l)
      | ^~~~~~
/usr/include/CGAL/Lazy.h:75:1: note:   template argument deduction/substitution failed:
/usr/include/CGAL/Lazy.h:121:100: note:   mismatched types ‘CGAL::Lazy<AT, ET, E2A>’ and ‘const int’
  121 | auto approx(T const& t) {return make_transforming_iterator(t,[](auto const&u){return CGAL::approx(u);});}
      |                                                                                                    ^
/usr/include/CGAL/Lazy.h:104:1: note: candidate: ‘const CGAL::Bbox_2& CGAL::approx(const CGAL::Bbox_2&)’
  104 | CGAL_LAZY_FORWARD(Bbox_2)
      | ^~~~~~~~~~~~~~~~~
/usr/include/CGAL/Lazy.h:104:1: note:   no known conversion for argument 1 from ‘const int’ to ‘const CGAL::Bbox_2&’
  104 | CGAL_LAZY_FORWARD(Bbox_2)
      | ^~~~~~~~~~~~~~~~~
/usr/include/CGAL/Lazy.h:105:1: note: candidate: ‘const CGAL::Bbox_3& CGAL::approx(const CGAL::Bbox_3&)’
  105 | CGAL_LAZY_FORWARD(Bbox_3)
      | ^~~~~~~~~~~~~~~~~
/usr/include/CGAL/Lazy.h:105:1: note:   no known conversion for argument 1 from ‘const int’ to ‘const CGAL::Bbox_3&’
  105 | CGAL_LAZY_FORWARD(Bbox_3)
      | ^~~~~~~~~~~~~~~~~
/usr/include/CGAL/Lazy.h:121:6: note: candidate: ‘template<class T, class> auto CGAL::approx(const T&)’
  121 | auto approx(T const& t) {return make_transforming_iterator(t,[](auto const&u){return CGAL::approx(u);});}
      |      ^~~~~~
/usr/include/CGAL/Lazy.h:121:6: note:   template argument deduction/substitution failed:
/usr/include/CGAL/Lazy.h: At global scope:
/usr/include/CGAL/Lazy.h:122:31: error: ‘enable_if_t’ in namespace ‘std’ does not name a template type; did you mean ‘enable_if’?
  122 | template <class T, class=std::enable_if_t<is_iterator_type<T,std::input_iterator_tag>::value>>
      |                               ^~~~~~~~~~~
      |                               enable_if
/usr/include/CGAL/Lazy.h:122:42: error: expected ‘>’ before ‘<’ token
  122 | template <class T, class=std::enable_if_t<is_iterator_type<T,std::input_iterator_tag>::value>>
      |                                          ^
In file included from /usr/include/CGAL/Lazy_exact_nt.h:33,
                 from /usr/include/CGAL/internal/Exact_type_selector.h:25,
                 from /usr/include/CGAL/Exact_kernel_selector.h:26,
                 from /usr/include/CGAL/Filtered_kernel.h:22,
                 from /usr/include/CGAL/Exact_predicates_inexact_constructions_kernel.h:21,
                 from ./trf.h:4,
                 from ./parser.h:10,
                 from ./eop_series.h:13,
                 from ./eop_series.cpp:1:
/usr/include/CGAL/Lazy.h:320:33: error: ‘std::index_sequence’ has not been declared
  320 |   void update_exact_helper(std::index_sequence<I...>) const {
      |                                 ^~~~~~~~~~~~~~
/usr/include/CGAL/Lazy.h:320:47: error: expected ‘,’ or ‘...’ before ‘<’ token
  320 |   void update_exact_helper(std::index_sequence<I...>) const {
      |                                               ^
/usr/include/CGAL/Lazy.h: In member function ‘void CGAL::Lazy_rep_n<AT, ET, AC, EC, E2A, L>::update_exact() const’:
/usr/include/CGAL/Lazy.h:327:30: error: ‘make_index_sequence’ is not a member of ‘std’
  327 |     update_exact_helper(std::make_index_sequence<sizeof...(L)>{});
      |                              ^~~~~~~~~~~~~~~~~~~
/usr/include/CGAL/Lazy.h:327:63: error: expected primary-expression before ‘{’ token
  327 |     update_exact_helper(std::make_index_sequence<sizeof...(L)>{});
      |                                                               ^
/usr/include/CGAL/Lazy.h: At global scope:
/usr/include/CGAL/Lazy.h:831:22: error: ‘remove_cv_t’ is not a member of ‘std’; did you mean ‘remove_cv’?
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                      ^~~~~~~~~~~
      |                      remove_cv
/usr/include/CGAL/Lazy.h:831:22: error: ‘remove_cv_t’ is not a member of ‘std’; did you mean ‘remove_cv’?
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                      ^~~~~~~~~~~
      |                      remove_cv
/usr/include/CGAL/Lazy.h:831:39: error: ‘remove_reference_t’ is not a member of ‘std’; did you mean ‘remove_reference’?
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                                       ^~~~~~~~~~~~~~~~~~
      |                                       remove_reference
/usr/include/CGAL/Lazy.h:831:89: error: template argument 1 is invalid
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                                                                                         ^~
/usr/include/CGAL/Lazy.h:831:22: error: ‘remove_cv_t’ is not a member of ‘std’; did you mean ‘remove_cv’?
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                      ^~~~~~~~~~~
      |                      remove_cv
/usr/include/CGAL/Lazy.h:831:22: error: ‘remove_cv_t’ is not a member of ‘std’; did you mean ‘remove_cv’?
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                      ^~~~~~~~~~~
      |                      remove_cv
/usr/include/CGAL/Lazy.h:831:39: error: ‘remove_reference_t’ is not a member of ‘std’; did you mean ‘remove_reference’?
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                                       ^~~~~~~~~~~~~~~~~~
      |                                       remove_reference
/usr/include/CGAL/Lazy.h:831:89: error: template argument 1 is invalid
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                                                                                         ^~
/usr/include/CGAL/Lazy.h:831:22: error: ‘remove_cv_t’ is not a member of ‘std’; did you mean ‘remove_cv’?
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                      ^~~~~~~~~~~
      |                      remove_cv
/usr/include/CGAL/Lazy.h:831:22: error: ‘remove_cv_t’ is not a member of ‘std’; did you mean ‘remove_cv’?
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                      ^~~~~~~~~~~
      |                      remove_cv
/usr/include/CGAL/Lazy.h:831:39: error: ‘remove_reference_t’ is not a member of ‘std’; did you mean ‘remove_reference’?
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                                       ^~~~~~~~~~~~~~~~~~
      |                                       remove_reference
/usr/include/CGAL/Lazy.h:831:89: error: template argument 1 is invalid
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                                                                                         ^~
/usr/include/CGAL/Lazy.h:831:22: error: ‘remove_cv_t’ is not a member of ‘std’; did you mean ‘remove_cv’?
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                      ^~~~~~~~~~~
      |                      remove_cv
/usr/include/CGAL/Lazy.h:831:22: error: ‘remove_cv_t’ is not a member of ‘std’; did you mean ‘remove_cv’?
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                      ^~~~~~~~~~~
      |                      remove_cv
/usr/include/CGAL/Lazy.h:831:39: error: ‘remove_reference_t’ is not a member of ‘std’; did you mean ‘remove_reference’?
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                                       ^~~~~~~~~~~~~~~~~~
      |                                       remove_reference
/usr/include/CGAL/Lazy.h:831:89: error: template argument 1 is invalid
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                                                                                         ^~
/usr/include/CGAL/Lazy.h:831:22: error: ‘remove_cv_t’ is not a member of ‘std’; did you mean ‘remove_cv’?
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                      ^~~~~~~~~~~
      |                      remove_cv
/usr/include/CGAL/Lazy.h:831:22: error: ‘remove_cv_t’ is not a member of ‘std’; did you mean ‘remove_cv’?
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                      ^~~~~~~~~~~
      |                      remove_cv
/usr/include/CGAL/Lazy.h:831:39: error: ‘remove_reference_t’ is not a member of ‘std’; did you mean ‘remove_reference’?
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                                       ^~~~~~~~~~~~~~~~~~
      |                                       remove_reference
/usr/include/CGAL/Lazy.h:831:89: error: template argument 1 is invalid
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                                                                                         ^~
/usr/include/CGAL/Lazy.h:831:3: error: invalid use of template-name ‘CGAL::Lazy_exact_nt’ without an argument list
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |   ^~~~~~~~~~~~~
/usr/include/CGAL/Lazy.h:831:3: note: class template argument deduction is only available with ‘-std=c++17’ or ‘-std=gnu++17’
In file included from /usr/include/CGAL/Lazy_exact_nt.h:33,
                 from /usr/include/CGAL/internal/Exact_type_selector.h:25,
                 from /usr/include/CGAL/Exact_kernel_selector.h:26,
                 from /usr/include/CGAL/Filtered_kernel.h:22,
                 from /usr/include/CGAL/Exact_predicates_inexact_constructions_kernel.h:21,
                 from ./trf.h:4,
                 from ./parser.h:10,
                 from ./eop_series.h:13,
                 from ./eop_series.cpp:1:
/usr/include/CGAL/Lazy.h:61:7: note: ‘template<class ET_> class CGAL::Lazy_exact_nt’ declared here
   61 | class Lazy_exact_nt;
      |       ^~~~~~~~~~~~~
In file included from /usr/include/CGAL/Lazy_exact_nt.h:33,
                 from /usr/include/CGAL/internal/Exact_type_selector.h:25,
                 from /usr/include/CGAL/Exact_kernel_selector.h:26,
                 from /usr/include/CGAL/Filtered_kernel.h:22,
                 from /usr/include/CGAL/Exact_predicates_inexact_constructions_kernel.h:21,
                 from ./trf.h:4,
                 from ./parser.h:10,
                 from ./eop_series.h:13,
                 from ./eop_series.cpp:1:
/usr/include/CGAL/Lazy.h:831:16: error: expected initializer before ‘<’ token
  831 |   Lazy_exact_nt<std::remove_cv_t<std::remove_reference_t<decltype(ec(CGAL::exact(l)...))>>>
      |                ^
In file included from /usr/include/CGAL/N_step_adaptor_derived.h:20,
                 from /usr/include/CGAL/Polyhedron_3.h:26,
                 from ./trf.h:6,
                 from ./parser.h:10,
                 from ./eop_series.h:13,
                 from ./eop_series.cpp:1:
/usr/include/CGAL/iterator.h:663:16: error: ‘decay_t’ in namespace ‘std’ does not name a template type; did you mean ‘decay’?
  663 |   typedef std::decay_t<typename cpp11::result_of<Op(arg_type)>::type> value_type;
      |                ^~~~~~~
      |                decay
/usr/include/CGAL/iterator.h:665:11: error: ‘value_type’ does not name a type; did you mean ‘arg_type’?
  665 |   typedef value_type const*                                     pointer;
      |           ^~~~~~~~~~
      |           arg_type
/usr/include/CGAL/iterator.h:666:11: error: ‘value_type’ does not name a type; did you mean ‘arg_type’?
  666 |   typedef value_type const&                                     reference;
      |           ^~~~~~~~~~
      |           arg_type
/usr/include/CGAL/iterator.h:671:11: error: ‘value_type’ does not name a type; did you mean ‘arg_type’?
  671 |   mutable value_type val;  // Note: mutable is needed because we want to
      |           ^~~~~~~~~~
      |           arg_type
/usr/include/CGAL/iterator.h:699:9: error: ‘value_type’ does not name a type; did you mean ‘arg_type’?
  699 |   const value_type& operator*() const {
      |         ^~~~~~~~~~
      |         arg_type
/usr/include/CGAL/iterator.h:715:9: error: ‘value_type’ does not name a type; did you mean ‘arg_type’?
  715 |   const value_type& operator[](difference_type i) const {
      |         ^~~~~~~~~~
      |         arg_type
/usr/include/CGAL/iterator.h:1321:44: error: ‘std::index_sequence’ has not been declared
 1321 | auto to_tuple(std::tuple<Args...> &t, std::index_sequence<Is...>)
      |                                            ^~~~~~~~~~~~~~
/usr/include/CGAL/iterator.h:1321:58: error: expected ‘,’ or ‘...’ before ‘<’ token
 1321 | auto to_tuple(std::tuple<Args...> &t, std::index_sequence<Is...>)
      |                                                          ^
/usr/include/CGAL/iterator.h:1321:1: error: ‘to_tuple’ function uses ‘auto’ type specifier without trailing return type
 1321 | auto to_tuple(std::tuple<Args...> &t, std::index_sequence<Is...>)
      | ^~~~
/usr/include/CGAL/iterator.h:1321:1: note: deduced return type only available with ‘-std=c++14’ or ‘-std=gnu++14’
/usr/include/CGAL/iterator.h: In member function ‘CGAL::Dispatch_output_iterator<std::tuple<_Tps ...>, std::tuple<_Args2 ...> >::operator std::tuple<O& ...>()’:
/usr/include/CGAL/iterator.h:1414:49: error: ‘index_sequence_for’ is not a member of ‘std’
 1414 |     return tuple_internal::to_tuple(*this, std::index_sequence_for<O...>{});
      |                                                 ^~~~~~~~~~~~~~~~~~
/usr/include/CGAL/iterator.h:1414:69: error: expected primary-expression before ‘...’ token
 1414 |     return tuple_internal::to_tuple(*this, std::index_sequence_for<O...>{});
      |                                                                     ^~~
/usr/include/CGAL/iterator.h: In member function ‘CGAL::Dispatch_output_iterator<std::tuple<_Tps ...>, std::tuple<_Args2 ...> >::operator std::tuple<const O& ...>() const’:
/usr/include/CGAL/iterator.h:1419:49: error: ‘index_sequence_for’ is not a member of ‘std’
 1419 |     return tuple_internal::to_tuple(*this, std::index_sequence_for<O...>{});
      |                                                 ^~~~~~~~~~~~~~~~~~
/usr/include/CGAL/iterator.h:1419:69: error: expected primary-expression before ‘...’ token
 1419 |     return tuple_internal::to_tuple(*this, std::index_sequence_for<O...>{});
      |                                                                     ^~~
In file included from ./ls_neq.h:4,
                 from ./sinex.h:10,
                 from ./trf.h:19,
                 from ./parser.h:10,
                 from ./eop_series.h:13,
                 from ./eop_series.cpp:1:
./ls_solution.h: In member function ‘virtual double ivg::Ls_solution::calc_posterior_vfac()’:
./ls_solution.h:85:9: warning: no return statement in function returning non-void [-Wreturn-type]
   85 |         };
      |         ^
./ls_solution.h: In member function ‘virtual double ivg::Ls_solution::calc_wrms()’:
./ls_solution.h:137:9: warning: no return statement in function returning non-void [-Wreturn-type]
  137 |         };
      |         ^
./ls_solution.h: In member function ‘virtual double ivg::Ls_solution::calc_rms()’:
./ls_solution.h:144:9: warning: no return statement in function returning non-void [-Wreturn-type]
  144 |         };
      |         ^
./ls_solution.h: In member function ‘virtual int ivg::Ls_solution::get_degrees_of_freedom()’:
./ls_solution.h:216:9: warning: no return statement in function returning non-void [-Wreturn-type]
  216 |         };
      |         ^
./ls_solution.h: In member function ‘virtual int ivg::Ls_solution::get_nobs()’:
./ls_solution.h:219:9: warning: no return statement in function returning non-void [-Wreturn-type]
  219 |         };
      |         ^
make: *** [Makefile:44: eop_series.o] Error 1
```
Led by notes in https://github.com/colmap/colmap/issues/793 I changed the MakeFile for ivg to allow compiling with libcgal5.

**Comment by varenius** [comment]
Fixed in 5ba4b8c21cf3f448ee40fbc44ae77a1249dff77a

**Comment by varenius** [comment]
Note: Also qtplot library fixed in 420a326bceffa3b5c6e9e0a037062bac8e512fe8.
